saleor/plugins/webhook/utils.py

```diff
@@ -0,0 +1,79 @@
+import re
+from typing import TYPE_CHECKING, List, Optional
+
+from ...payment.interface import GatewayResponse, PaymentGateway, PaymentMethodInfo
+
+if TYPE_CHECKING:
+    from requests.models import Response as RequestsResponse
+
+    from ...app.models import App
+    from ...payment.interface import PaymentData
+
+
+def to_payment_app_id(app: "App", gateway_id: str) -> "str":
+    return f"app:{app.pk}:{gateway_id}"
+
+
+def from_payment_app_id(app_gateway_id: str) -> Optional["int"]:
+    pattern = r"^app:(?P<app_pk>[0-9]+):[-a-zA-Z0-9_]+"
+    match = re.match(pattern, app_gateway_id)
+    app_pk = None
+    if match:
+        app_pk = match.groupdict().get("app_pk")
+    return int(app_pk) if app_pk else None
+
+
+def webhook_response_to_payment_gateways(
+    response: "RequestsResponse", app: "App"
+) -> List["PaymentGateway"]:
+    response_json = response.json()
+    gateways = []
+    for gateway_data in response_json:
+        gateway_id = gateway_data.get("id")
+        gateway_name = gateway_data.get("name")
+        gateway_currencies = gateway_data.get("currencies")
+        gateway_config = gateway_data.get("config")
+
+        gateways.append(
+            PaymentGateway(
+                id=to_payment_app_id(app, gateway_id),
+                name=gateway_name,
+                currencies=gateway_currencies,
+                config=gateway_config,
+            )
+        )
+    return gateways
+
+
+def webhook_response_to_gateway_response(
+    payment_information: "PaymentData",
+    response: "RequestsResponse",
+    transaction_kind: "str",
+) -> "GatewayResponse":
+    response_json = response.json()
+
+    error = response_json.get("error")
+    is_success = response.status_code == 200 and not error
+
+    payment_method_info = PaymentMethodInfo(
```

Comment:
I would consider to place all `PaymentMethodInfo` data to json separate object. Something like this:
```
response = {
  payment_method_info: {
      ...
  }
  ```

Reply:
It is a separate object, it is returned exactly as you proposed. Or I don't understand the comment.

Reply:
I was meaning that now you have flat response object, like:
```
response = response.json()
assert response == {
...
kind: capture,
error: None, 
payment_method_brand: "something",
payment_method_exp_month: "something2",
payment_method_last_4:"something3"
...
}
```
And my proposition was that, maybe we could move all payment_method data in separate json data. So your response from payment gateway would be:
```
response = response.json()
assert response == {
...
kind: capture,
error: None,
payment_method:{ 
    brand: "something",
    exp_month: "something2",
    last_4:"something3"
}
...
}
```

Reply:
Ah, you meant the incoming response, not the payload - we can change it :)